### ISSUE TITLE
_yarn_linear_ramp_mask Seems Not Calculated as Expected

### ISSUE THREAD

**Issue by siyuanseever** [jquesnelle/yarn#65]
Below is the simplified implementation and usage of `_yarn_linear_ramp_mask`:

```python
def _yarn_linear_ramp_mask(min, max, dim):
    linear_func = (torch.arange(dim, dtype=torch.float32) - min) / (max - min)
    ramp_func = torch.clamp(linear_func, 0, 1)
_yarn_linear_ramp_mask(low, high, self.dim // 2)
```

However, I believe the correct implementation should be:

```python
def _yarn_linear_ramp_mask(min, max, dim):
    linear_func = (torch.arange(0, dim, 2, dtype=torch.float32) - min) / (max - min)
    ramp_func = torch.clamp(linear_func, 0, 1)
_yarn_linear_ramp_mask(low, high, self.dim)
```

This is because the coordinates of `low` and `high` are based on `dim`, and the range should be `[0, self.dim - 2]`.
